### PR TITLE
Adding stdin and stdout support

### PR DESCRIPTION
**Pull request by zyronix**
# Stdin / Stdout support
## Description
This pull requests adds support for stdin support and stdout support. It changes quite some behavior aswell.

## Changes
 - By default, when no --input is specified, stdin will be used.
 - By default, when no --ouput is specified, stdout will be used (thus printing results to stdout).
 - By default when logfile is specified, only lines that break/are removed are now added to this file. 
 - This same behavior can be enabled for the commandline using --verbose
 - To enable the old behavior to log all changes, use the --debug option. This will print or write all changes made to any line.

## Usage
```
hashcat rockyou.txt --stdout -r best64.txt | demeuk --check-min-length 5 --check-max-length 32 -j all | sort -u
```

### CODE REVIEW

**Comment by zyronix** [comment]
@rixvet Could you review this? It is bit experimental, but should work.
What do you think of this feature?
If no --input is given, should it go to stdin by default?

**Review by rixvet — COMMENTED** [review]
Reviewed the code and start playing around removing duplicate code. Turned out to be bit much to show in review comments. Thus started a new branch for illustration purposes for more details see https://github.com/NetherlandsForensicInstitute/demeuk/pull/50

File: bin/demeuk.py [hunk 1]
```diff
@@ -14,18 +14,21 @@
...
-        -o --output <path to file>      Specify the output file name.
-        -l --log <path to file>         Optional, specify where the log file needs to be writen to
-        -j --threads <threads>          Optional, demeuk doesn't use threads by default. Specify amount of threads to
-                                        spawn. Specify the string 'all' to make demeuk auto detect the amount of threads
-                                        to start based on the CPU's.
+        -i --input <path to file>       Specify the input file to be cleaned, or provide a glob pattern.
+                                        (default: stdin)
+        -o --output <path to file>      Specify the output file name. (default: stdout)
+        -l --log <path to file>         Optional, specify where the log file needs to be writen to (default: stderr)
+        -j --threads <threads>          Optional, specify amount of threads to spawn. Specify the string 'all' to make
```
Comment (GingerGeneste) [inline bin/demeuk.py:24]: By default using all cores is more intuitive.

**Review by GingerGeneste — APPROVED** [review]
Looks fine, we discussed already a few points.
Two additions in the review:
- Dont forget to update the readme as well (docs are fine)
- Dont forget to check if the input file exists / let the script throw an error if it doesnt.

**Pull request merged**